crates/ruff_python_formatter/README.md

```diff
@@ -323,3 +323,188 @@ await [1, 2, 3]
 This is more consistent to the formatting of other awaited expressions: Ruff and Black both
 remove parentheses around, e.g., `await (1)`, only retaining them when syntactically required,
 as in, e.g., `await (x := 1)`.
+
+### Line breaks between implicitly concatenated string attributes are not preserved ([#7052](https://github.com/astral-sh/ruff/issues/7052))
+
+Black preserves line breaks between implicitly concatenated string attributes.
+
+```python
+print(
+    "aaaaaaa {:.2f}% "
+    "aaaaaaaaaaaaaaaa".format(
+        (
+            ((df_aaaaaaaaaaa["key"].isna()) | (df_aaaaaaaaaaa["key"] == 0)).sum()
+            / len(df_aaaaaaaaaaa)
+        )
+        * 100
+    )
+)
+```
+
+Ruff will instead collapse them, as long as the resulting line does not exceed the configured
+line width:
+
+```python
+print(
+    "aaaaaaa {:.2f}% " "aaaaaaaaaaaaaaaa".format(
+        (
+            ((df_aaaaaaaaaaa["key"].isna()) | (df_aaaaaaaaaaa["key"] == 0)).sum()
+            / len(df_aaaaaaaaaaa)
+        )
+        * 100
+    )
+)
+```
+
+### Own-line comments on expressions don't cause the expression to expand ([#7314](https://github.com/astral-sh/ruff/issues/7314))
+
+Given an expression like:
+
+```python
+(
+    # A comment in the middle
+    some_example_var and some_example_var not in some_example_var
+)
+```
+
+Black associates the comment with `some_example_var`, thus splitting it over two lines:
+
+```python
+(
+    # A comment in the middle
+    some_example_var
+    and some_example_var not in some_example_var
+)
+```
+
+Ruff will instead associate the comment with the entire boolean expression, thus preserving the
+initial formatting:
+
+```python
+(
+    # A comment in the middle
+    some_example_var and some_example_var not in some_example_var
+)
+```
+
+### Tuples in assignments are parenthesized when expanded ([#7317](https://github.com/astral-sh/ruff/issues/7317))
```

Comment:
I think this applies to all tuples, not just in assignment expressions (or maybe all tuples except in for positions?)

```python
(0.0, 0.0), (0.0, 0.0,)

# Ruffed
(
    (0.0, 0.0),
    (
        0.0,
        0.0,
    ),
)

# Blacked
(0.0, 0.0), (
    0.0,
    0.0,
)
```

There's also another difference where Ruff always parenthesizes single element tuples where Black does not

```python
(0.0, 0.0),

# Ruffed
((0.0, 0.0),)

# Blacked
(0.0, 0.0),
```

This is to avoid accidental tuple expressions (e.g. you delete the second element but forget to remove the `,`)

Reply:
Thank you. I moved the single-element tuple thing to its own deviation.